### DISCUSSION TITLE
Content missing from the docs

### DISCUSSION THREAD

**Discussion by pld** [opensrp/fhircore#1978] [Ideas]
This is a discussion to record anything missing from the [docs](https://fhircore.smartregister.org/). Also feel free to submit a PR and update the docs directly. But if you don't have time for that, or are not sure where in the docs it goes but know it should be in the docs, please add a thread here.

Below is an aggregated task list of docs that are remaining to add, we can check them off as we've merged PRs for them.

## RBAC (Role Based Access)
- [ ] User management - FHIRWeb/Keycloak
- [ ] Permission Access checker - Design/Implementation/Configuration
- [ ] Data Filtering/Sync strategy - Design/Implementation/Configuration
- [ ] Architecture - Diagram that includes Client, Access Proxy, Keycloak, FHIRWeb, Backend and their interactions
- [ ] Server deployment - Architecture + Deployment Steps/Workflow

## How to add/modify flavors, including
- [x] what all the keys in the flavor config entry mean
- [x] when to add and not to add a new flavor
- [ ] guidance on whether we'll merge PRs with new flavors into main

## Form authoring
- [x] Make sure the `enableWhen` expressions are accompanied by the `enableBehaviour` attribute in order to avoid form crashes. 
    - The `enableBehaviour` defines whether the expressions on the `enableWhen` are to use `and` or `or` logic. 
    - The `enableBehaviour` options are `any` for `or` & `all` for `and`
- [x] Here is a [sample](https://github.com/opensrp/fhircore/discussions/1978#discussioncomment-4839319) `enableWhen` and `enableBehaviour` attribute. 

## Multi-Language Support
- [ ] How to add Language configs to the composition resource 
- [ ] How to add Langage Binaries to HAPI FHIR Server 
- [ ] How to load String.xml files to the android assets folder 
- [ ] Publishing translation using Transifex

**Comment by pld** [comment]
Add docs on how to add/modify flavors, including
- what all the keys in the flavor config entry mean
- when to add and not to add a new flavor
- guidance on whether we'll merge PRs with new flavors into main

h/t @f-odhiambo

**Reply by sharon2719** [reply]
@pld Can I pick up on this?

**Reply by pld** [reply]
Yes, sounds great! Please create a new issue with the specific docs you're planning on adding, then comment it with your expected timeline (if it goes longer than the initial expected, no problem, just add another comment with the updated timeline), then you can open a PR against that issue and tag me for review. Thank you!

**Comment by ageryck** [comment]
To include these contents later

**Category**: Form authoring (Things to look out for when writing your forms)

- Make sure the `enableWhen` expressions are accompanied by the `enableBehaviour` attribute in order to avoid form crashes. 
    - The `enableBehaviour` defines whether the expressions on the `enableWhen` are to use `and` or `or` logic. 
    - The `enableBehaviour` options are `any` for `or` & `all` for `and`
- Here is a sample `enableWhen` and `enableBehaviour` attribute.
```json
{
  "enableWhen": [
    {
      "question": "cf0093a9-1e38-422b-fc81-273caa0bb48a",
      "operator": "=",
      "answerCoding": {
        "system": "urn:uuid:d3051c21-8668-412e-a2dd-105cf274b260",
        "code": "no"
      }
    },
    {
      "question": "94357450-e603-49f7-899c-464ecd10ddd1",
      "operator": "=",
      "answerCoding": {
        "system": "urn:uuid:4ae66618-142d-4e2c-dd66-5d63c14b83a9",
        "code": "no"
      }
    },
    {
      "question": "14648d8c-c88f-40be-9f25-66d7ba621482",
      "operator": "=",
      "answerCoding": {
        "system": "urn:uuid:08516020-d926-4463-863e-d6f45b0d317b",
        "code": "no"
      }
    }
  ],
  "enableBehaviour":"all"
}
```

**Reply by pld** [reply]
Can you edit above and paste the content  please?

**Reply by sharon2719** [reply]
@pld can I probably pick up on this?

**Reply by pld** [reply]
awesome, yes, that'd be great!

**Comment by ndegwamartin** [comment]
**Documentation required for RBAC (Role Based Access)** 
- [ ] User management - FHIRWeb/Keycloak
- [ ] Permission Access checker - Design/Implementation/Configuration
- [ ] Data Filtering/Sync strategy - Design/Implementation/Configuration
- [ ] Architecture - Diagram that includes Client, Access Proxy, Keycloak, FHIRWeb, Backend and their interactions
- [ ] Server deployment - Architecture + Deployment Steps/Workflow

**Reply by pld** [reply]
Cool, good call, I'm gonna start aggregating stuff into a task list at the top

**Comment by f-odhiambo** [comment]
Documentation required for Multi-Language Support

- [ ] How to add Language configs to the composition resource 
- [ ] How to add Langage Binaries to HAPI FHIR Server 
- [ ] How to load String.xml files to the android assets folder 
- [ ] Publishing translation using Transifex

**Reply by sharon2719** [reply]
@pld can I pick up on this?

**Comment by ellykits** [comment]
#### Rules engine library on maintenance mode

The [Rules Engine library](https://github.com/j-easy/easy-rules) we are using is currently in maintenance mode. The version is compatible with JEXL libraries v3.1...v3.2.1, but cannot work with the latest version v3.3 of JEXL.

We could explore the following options:

1. Fork the repository to opensrp account
2. Identify and use any of the forks of the Rules Engine library that is under active development
3. Seek for an alternative rules engine library

**Reply by pld** [reply]
Can you add a link to the rules engine library we're using please
